components/reusable-components/tags-modal/TagsModal.tsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+
+'use client';
+
+import React, { ChangeEvent, useState } from 'react';
+import styles from './TagsModal.module.scss';
+
+interface TagSearchProps {
+  existingTags: string[];
+  selectedTags: string[];
+  setSelectedTags: (items: string[]) => void;
+  onClose: () => void;
+}
+
+const TagsModal = ({ existingTags, selectedTags, setSelectedTags, onClose }: TagSearchProps) => {
+  const [searchTag, setSearchTag] = useState<string>('');
+  const [filteredTags, setFilteredTags] = useState<string[]>([]);
+
+  const handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const query = e.target.value;
+    setSearchTag(query);
+
+    if (query) {
+      const filtered = existingTags.filter((tag) =>
+        tag.toLowerCase().includes(query.toLowerCase())
+      );
+      setFilteredTags(filtered);
+    } else {
+      setFilteredTags([]);
+    }
+  };
```

Comment:
We should use debounce on the search input. So we can improve performance and avoid sending requests on every keystroke.

Reply:
The more I think about it, this should be implemented as following.
We will fetch the tags list, and memoize it. Search through that list, if we don't find that item we can send a request to add the tag.